### DISCUSSION TITLE
MIME type for .fgb

### DISCUSSION THREAD

**Discussion by rressler** [flatgeobuf/flatgeobuf#112] [Q&A]
Does anyone know the status of obtaining a MIME type for .fgb files? PanDoc does not support .fgb yet and needs a MIME type and description to do so 
Thanks

**Comment by bjornharrtell** [comment]
No progress. The process seem very involved and complicated. To me much more difficult than creating flatgeobuf in the first place. 😂 I've asked around for help but haven't got any yet.

PS. It seems to me quite backwards/stagnant of pandoc to require an existing mimetype.

**Reply by bjornharrtell** [reply]
Also, can you clarify why pandoc is relevant at all? It seems to be a markup language converter.

**Reply by tim-salabim** [reply]
I can only speak for the R language, it is very important for report generation. So relevant for industry, academia, ...

**Comment by rressler** [comment]
I use pandoc to generate HTML files from R Markdown files. I was trying to include a mapview package interactive map that was generating a .fgb file but pandoc did not recognize it so errored out.
Thanks for the update.
Richard
From: Björn Harrtell ***@***.***>
Reply-To: flatgeobuf/flatgeobuf ***@***.***>
Date: Sunday, April 4, 2021 at 2:10 PM
To: flatgeobuf/flatgeobuf ***@***.***>
Cc: Richard Ressler ***@***.***>, Author ***@***.***>
Subject: Re: [flatgeobuf/flatgeobuf] MIME type for .fgb (#112)


External Email: Use caution with links and attachments.

Also, can you clarify why pandoc is relevant at all? It seems to be a markup language converter.

**Comment by bjornharrtell** [comment]
This will possibly be a lesser issue after the resolution of https://github.com/jgm/pandoc/issues/7202 has been released and used in dependent projects.

**Comment by m-mohr** [comment]
It would still be interesting to get a media type for the format, it is very useful in HTTP world as well. We use them also a lot in STAC.
The most reasonable thing is probably to just define one in the vendor prefixed area: https://en.wikipedia.org/w/index.php?title=Media_type#Registration_trees
So for example `application/vnd.flatgeobuf`.

**Reply by bjornharrtell** [reply]
Ping @cholmes and @ogcscotts. :D

**Reply by gadomski** [reply]
If it makes sense, we'd like to add flatgeobuf as a built-in media type in PySTAC: https://github.com/stac-utils/pystac/pull/938. The theory is that building a value into PySTAC will help consistent adoption by downstream STAC producers, and therefore help uptake of the "correct" MIME type. However, as @m-mohr correctly pointed out, we shouldn't just pick one. @bjornharrtell do you have a value that we _should_ use in PySTAC for media type (e.g. `application/vnd.flatgeobuf`)? Or is it to soon to settle on one?

**Reply by bjornharrtell** [reply]
I have aimed for application/flatgeobuf but perhaps it is never going to happen, who knows? If application/vnd.flatgeobuf is easier to get why not. But I'm not going to spend time on it because it's just too boring/fruitless for me.

**Reply by cholmes** [reply]
It's typical to evolve from application/vnd. to application/ - and typically you wouldn't use application/ until it's fully approved. So I'd say just declare informally here (or ideally in the repo somewhere) that it's `application/vnd.flatgeobuf`. Ideally we'd submit that, but I don't think it's really necessary. And then we wait for @ogcscotts to do his OGC magic to get application/flatgeobuf.

**Reply by ogcscotts** [reply]
I owe @bjornharrtell and @cholmes the final approval steps for the OGC work item - my slowness entirely (will deal with in email thread). Once we get this into the process, I am happy to do the MIME application.

**Reply by m-mohr** [reply]
Is there any update on this?

**Reply by ogcscotts** [reply]
getting there.... we had to restart the vote on the work item due to a slow holiday season in December. I expect that we can start the internal review and final approval in mid-March.

**Reply by bjornharrtell** [reply]
hey @ogcscotts any news? :)

**Reply by ogcscotts** [reply]
@bjornharrtell no good news, sorry - we did not get enough voters. However, this also fell into a time where the Community Standard process was being changed and one other effort is in a holding pattern. If you have the appetite, we can work together in the revised process.
